addons/sourcemod/scripting/myjailshop.sp

```diff
@@ -2649,22 +2673,56 @@ void Item_Wallhack(int client, char[] name)
 		{
 			g_bWallhack[client] = true;
 			
-			for (int i = 1; i <= MaxClients; i++) if (IsClientInGame(i)) Setup_WallhackSkin(i);
+			for (int i = 1; i <= MaxClients; i++)if (IsClientInGame(i))Setup_WallhackSkin(i);
 			
-			CreateTimer (gc_fWallhackTime.FloatValue, Timer_Wallhack, client);
+			CreateTimer(gc_fWallhackTime.FloatValue, Timer_Wallhack, client);
 			
-			Forward_OnSetCredits(client,(Forward_OnGetCredits(client)-gc_iWallhack.IntValue));
+			Forward_OnSetCredits(client, (Forward_OnGetCredits(client) - gc_iWallhack.IntValue));
 			
 			Forward_OnPlayerBuyItem(client, name);
 			
 			CPrintToChat(client, "%t %t", "shop_tag", "shop_wallhack", client);
-			if (gc_bLogging.BoolValue) LogToFileEx(g_sPurchaseLogFile, "Player %L bought: Wallhack", client);
+			if (gc_bLogging.BoolValue)LogToFileEx(g_sPurchaseLogFile, "Player %L bought: Wallhack", client);
 		}
 		else CPrintToChat(client, "%t %t", "shop_tag", "shop_missingcredits", Forward_OnGetCredits(client), gc_iWallhack.IntValue);
 	}
 	else CPrintToChat(client, "%t %t", "shop_tag", "shop_alive");
 }
 
+void Item_RandomTP(int client, char[] name)
+{
+	if (IsPlayerAlive(client))
+	{
+		float fClientOrigin[3];

```

Comment:
here we should add an check for enough credits. like:
`if (Forward_OnGetCredits(client) >= gc_iRandomTP.IntValue)`
to prevent glitching.  
I see, I have also forgot to implement this check on some items... maybe you can complete it? if not, i will do it